drivers/storage/portworx/components_test.go

```diff
@@ -12815,20 +12813,15 @@ func TestPodDisruptionBudgetWithMetroDR(t *testing.T) {
 	expectedClusterDomainsEnumerateResp := &osdapi.SdkClusterDomainsEnumerateResponse{
 		ClusterDomainNames: []string{"domain 1", "domain 2"},
 	}
-	// StorageNode with ID 1,2,3 will run on the cluster
-	// k8s node4 doesn't meet affinity and node5 is tainted
-	// StorageNode with ID 6 and 7 don't belong to this cluster
+
+	// Case 1: When all nodes in the cluster use portworx version 3.1.1
 	expectedNodeEnumerateResp := &osdapi.SdkNodeEnumerateWithFiltersResponse{
 		Nodes: []*osdapi.StorageNode{
-			{Pools: []*osdapi.StoragePool{{ID: 1}}, SchedulerNodeName: "node1", ClusterDomain: "domain 1"},
-			{Pools: []*osdapi.StoragePool{{ID: 2}}, SchedulerNodeName: "node2", ClusterDomain: "domain 1"},
-			{Pools: []*osdapi.StoragePool{{ID: 3}}, SchedulerNodeName: "", Hostname: "node3", ClusterDomain: "domain 1"},
-			{Pools: []*osdapi.StoragePool{{ID: 4}}, SchedulerNodeName: "node4"},
-			{Pools: []*osdapi.StoragePool{{ID: 5}}, SchedulerNodeName: "node5"},
-			{Pools: []*osdapi.StoragePool{{ID: 6}}, SchedulerNodeName: "metro-dr-node1", ClusterDomain: "domain 2"},
-			{Pools: []*osdapi.StoragePool{{ID: 7}}, SchedulerNodeName: "metro-dr-node2", ClusterDomain: "domain 2"},
-			{Pools: []*osdapi.StoragePool{}},
-			{},
+			{Pools: []*osdapi.StoragePool{{ID: 1}}, SchedulerNodeName: "node1", ClusterDomain: "domain 1", NodeLabels: map[string]string{pxutil.NodeLabelPortworxVersion: "3.1.1"}},
+			{Pools: []*osdapi.StoragePool{{ID: 2}}, SchedulerNodeName: "node2", ClusterDomain: "domain 1", NodeLabels: map[string]string{pxutil.NodeLabelPortworxVersion: "3.1.1"}},
+			{Pools: []*osdapi.StoragePool{{ID: 3}}, SchedulerNodeName: "", Hostname: "node3", ClusterDomain: "domain 1", NodeLabels: map[string]string{pxutil.NodeLabelPortworxVersion: "3.1.1"}},
+			{Pools: []*osdapi.StoragePool{{ID: 4}}, SchedulerNodeName: "node4", ClusterDomain: "domain 2", NodeLabels: map[string]string{pxutil.NodeLabelPortworxVersion: "3.1.1"}},
+			{Pools: []*osdapi.StoragePool{{ID: 5}}, SchedulerNodeName: "node5", ClusterDomain: "domain 2", NodeLabels: map[string]string{pxutil.NodeLabelPortworxVersion: "3.1.1"}},
```

Comment:
Can we add a decommissioned node with an older version here, so we can test the case that we ignore such nodes?